Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,212 @@
+
+  /// Parses variable keys and their ranges from a formatKey.
+  /// Every variable key that is contained within the format string is preceded
+  /// by the %#@ characters or by the positional variants, e.g. %1$#@, and followed by the @ character.
+  ///
+  /// - Parameter formatKey: The formatKey from which the variable keys should be parsed.
+  /// - Returns: An array of discovered variable keys and their range within the `formatKey`.
+  // swiftlint:disable:next discouraged_optional_collection
+  private static func variableKeysFromFormatKey(_ formatKey: String) -> [VariableKeyResult]? {
+    let pattern = #"(%(?>(\d)\$)?#@([\w\.\p{Pd}]+)@)"#
+    guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else { return nil }
+    let nsrange = NSRange(formatKey.startIndex..<formatKey.endIndex, in: formatKey)
+    let matches = regex.matches(in: formatKey, options: [], range: nsrange)
+
+    return matches.compactMap { match -> (String, Range<String.Index>, Int?)? in
+      // 1st capture group is the whole format string including delimeters
+      let rangeNSRange = match.range(at: 1)
+      // 2nd capture group is the positional argument of the format string (Optional!)
+      let positionalArgumentNSRange = match.range(at: 2)
+      // 3rd capture group is the key, the string in between the delimeters
+      let keyNSRange = match.range(at: 3)
+
+      guard let rangeRange = Range(rangeNSRange, in: formatKey) else { return nil }
+      guard let keyRange = Range(keyNSRange, in: formatKey) else { return nil }
+      guard let positionalArgumentRange = Range(positionalArgumentNSRange, in: formatKey) else {
+        return (String(formatKey[keyRange]), rangeRange, nil)
+      }
+
+      return (String(formatKey[keyRange]), rangeRange, Int(formatKey[positionalArgumentRange]))
+    }
+  }
+}
+
+extension StringsDict.PluralEntry {
+  var translation: String? {
+    var result = formatKey
+
+    for intermediateResult in sequence(first: result, next: unfurlVariableKeysInFormatKey(_:)) {
+      result = intermediateResult
+    }
+
+    return "Plural case 'other': \(result)"
+  }
+
+  /// Unfurls any variable keys in a formatKey to get a possible translation for the `other` case.
+  ///
+  /// This method should be used in a recursive context, in which the output will be used as the input for the
+  /// next iteration.
+  ///
+  /// - Parameter formatKey: A format key containing variables.
+  /// - Returns: The format key in which all its variable keys are replaced with the content of the `other` translation,
+  ///  if the `formatKey` contained any variable keys, `nil` otherwise.
+  private func unfurlVariableKeysInFormatKey(_ formatKey: String) -> String? {
+    guard let firstRemainingVariableKey = StringsDict.variableKeysFromFormatKey(formatKey)?.first else {
+      return nil
+    }
+
+    let (key, range, _) = firstRemainingVariableKey
+    guard let variable = variables.first(where: { $0.key == key }) else { return nil }
```

Comment:
🤔  Shouldn't we emit a `logMessage(.warning, …)` here to tell that we found a variable in the `formatKey` that was not in the defined `variables`?

Reply:
EDIT: nvm, that seems to be taken care of upstream [in the StringsDictFileParser](https://github.com/SwiftGen/SwiftGen/pull/634/files#diff-00b60f7144570710db30567b97c39466R50-R55) so I think we're good.